### PR TITLE
Migrate to 1es templates for official build

### PR DESCRIPTION
**Pull request by dibarbet**
Passing build - https://dnceng.visualstudio.com/internal/_build/results?buildId=2407671&view=results
Passing release - https://dnceng.visualstudio.com/internal/_build/results?buildId=2407724&view=logs&j=82be16ef-8a99-5e32-720d-0cb728275b7b&t=82be16ef-8a99-5e32-720d-0cb728275b7b

### CODE REVIEW

File: azure-pipelines/build-all.yml [hunk 1]
```diff
@@ -2,24 +2,94 @@ parameters:
...
+      ${{ if eq(parameters.isOfficial, true) }}:
+        name: netcore1espool-internal
+        image: 1es-ubuntu-2204-pt
+      ${{ else }}:
+        name: Azure Pipelines
+        vmImage: ubuntu-latest
+      os: linux
+    steps:
+    - pwsh: |
+        $isPrerelease = $true
```
Comment (dibarbet) [inline azure-pipelines/build-all.yml:30]: was having some issues with uploading both release and prerelease artifacts.  Decided to change it slightly to instead figure out release or prerelease based on the branch, with an optional override.

If we need a release build from not the release branch, its easy to queue up another build with the channel override set.

File: azure-pipelines/release.yml [hunk 2]
```diff
@@ -1,131 +1,272 @@
...
+                  project: 'internal'
+                  definition: 1264
+                  buildVersionToDownload: 'specific'
+                  buildId: '$(resources.pipeline.officialBuildCI.runID)'
+                  branchName: '$(resources.pipeline.officialBuildCI.sourceBranch)'
+              - pwsh: |
+                  npm install --global vsce
+                displayName: 'Install vsce'
+              - pwsh: |
+                  # Choose whether to upload to prerelease or release based on the artifacts generated by the build pipeline.
```
Comment (dibarbet) [inline azure-pipelines/release.yml:62]: Release pipeline now automatically detects the correct channel based on the build artifact name

File: azure-pipelines-official.yml [hunk 3]
```diff
@@ -5,17 +5,39 @@ trigger:
     - release
```
Comment (dibarbet) [inline azure-pipelines-official.yml:5]: view with whitespace off

File: azure-pipelines-official.yml [hunk 4]
```diff
@@ -5,17 +5,39 @@ trigger:
...
-  # Useful when we need to create an updated extension version with no changes (for example to fix a bad partial release).
-  - name: versionNumberOverride
-    type: string
-    default: 'default'
+# Allows the version number to be set manually for a specific build.
+# Useful when we need to create an updated extension version with no changes (for example to fix a bad partial release).
+- name: versionNumberOverride
+  type: string
+  default: 'default'
+# Allows the marketplace release channel to be set manually for a specific build
```
Comment (dibarbet) [inline azure-pipelines-official.yml:14]: view PR with whitespace off

**Review by arkalyanms — APPROVED** [review]

**Pull request merged**